### ISSUE TITLE
Validation set approach as method in trainControl?

### ISSUE THREAD

**Issue by gsverhoeven** [topepo/caret#1019]
When learning resampling methods, students are often first exposed to what is often called the validation set approach (see for example chapter 5 of "An introduction to statistical learning"). In this approach, one simply splits the data at random in two parts, fits the model on one part and evaluates on the held-out part. 

It would be nice, primarily for teaching purposes, to have an easy way to use the validation set approach in caret. This could also serve as a first introduction to the caret package.

My impression is that such functionality currently does not exist.

I am aware of the possibility of combining `createDataPartition` function with  the  `index` argument of `trainControl` , to implement this (I wrote it up as a blog post).

However, this is not really suitable as a first introduction to caret. When using the `index` argument, the `method` argument is ignored, which is confusing for beginners. It also creates stratified samples, that adds further complexity that is unwanted in a teaching setting. 

So i am proposing creating a new method = "validation" that creates random samples (not stratified) and performs the validation set approach (this is not the same as 2-fold cv). 
 
I could write a pull request if the maintainers of caret will allow such and extension.

**Comment by topepo** [comment]
You can do a single iteration of Monte Carlo CV (aka leave-group-out CV) using 

```r
trainControl(method = "LGOCV", number = 1, p)
```

`p` controls how much goes into the training set (and `1-p` to the validation set)

**Comment by gsverhoeven** [comment]
Ah, i thought that was for nested data.

I tried it out, it indeed fits a model on fraction `p` of the data and predicts on the held out part.
But i cannot get it to run multiple times. I tried `repeats` but it gave me `"repeats" has no meaning for this resampling method.`

For teaching / learning about resampling methods, it would be nice to compare repeated validation set approach with e.g. repeated k-fold CV. For example as a function of fraction `p` used to control the relative size of train and test set.

**Comment by topepo** [comment]
So this would use a single split of the training set and divide it into a data set used for modeling and another for validation (e.g. performance assessment). If you want to do this multiple times, you can adjust `number`. 

Can you give a diagram or something for what you are looking for (if that's not it)?

**Comment by gsverhoeven** [comment]
Doh! I still kept thinking the number was for choosing the number of groups and setting it to 1 was required to make it hold out a single group. Sorry for not getting this.

LGOCV with `number` to set the number of repeats was exactly what i was looking for / proposing to implement. 

From a few posts on stackoverflow, I get the impression i'm not the only one that got confused by the naming here. You could consider adding a method `validation` or `train_test` with parameter `repeats` that simply maps to `LGOCV` with parameter `number` and keep those around as well for backwards compatibility.
But maybe that's confusing too .... 

For me the issue is solved at least.

Thanks!!!!
